pkg/kubelet/kubelet.go

```diff
@@ -2421,6 +2424,46 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}
 }
 
+func (kl *Kubelet) syncOSAndArchLabelsOnce() (*v1.Node, error) {
+	node, err := kl.GetNode()
+	if err != nil {
+		return nil, fmt.Errorf("error getting node: %v", err)
+	}
+	newNode := node.DeepCopy()
+	// Set the OS label if there is mismatch
+	osName, ok := newNode.Labels[v1.LabelOSStable]
+	if !ok || osName != sysruntime.GOOS {
+		if len(newNode.Labels) == 0 {
```

Comment:
Should we log a warning here if the labels are not in sync?
I think people might be interested if something is repeatedly changing node labels to incorrect values.

Reply:
I don't have a strong opinion one way or the other. The logs could cause unncessary spamming sometimes. However I can't think of an entity which changes the values apart from kubelet unless it is sort of rogue entity and which should be caught by other means like OPA or validating admission hooks.